### PR TITLE
RATE-4077 - Fixed TE message class errors

### PR DESCRIPTION
**Pull request by yangwong9664**

### CODE REVIEW

File: app/preview/TemplateParams.scala [hunk 1]
```diff
@@ -741,6 +741,36 @@ object TemplateParams {
...
+      "receivedTime" -> "21:45", "receivedDate" -> "20 April 2017", "identifier" -> "12345",
+      "subject" -> "subject",
+      "errorType" -> "fatal",
+      "errorNumber" -> "123", "errorMessage" -> "This is an error"
+    ),
+    "transactionEngineHMRCPSAPRACDEAUTHSuccess" -> Map(
+      "receivedTime" -> "21:45", "receivedDate" -> "20 April 2017", "identifier" -> "12345",
+      "subject" -> "subject"
+    ),
+    "transactionEngineHMRCPSAPRACDEAUTHFailure" -> Map(
```
Comment (enriquefaci) [inline app/preview/TemplateParams.scala]: transactionEngineHMRCPSAPRACDEAUTHFailure is added twice. Could you correct it, please?

File: app/uk/gov/hmrc/hmrcemailrenderer/templates/transactionengine/TransactionEngineTemplates.scala [hunk 2]
```diff
@@ -890,6 +908,42 @@ object TransactionEngineTemplates {
...
+          plainTemplate = psa.prac.txt.transactionEngineHMRCPSAPRACAUTHSuccess.f,
+          htmlTemplate = psa.prac.html.transactionEngineHMRCPSAPRACAUTHSuccess.f,
+          priority = Some(MessagePriority.Urgent)
+        ),
+        MessageTemplate.createWithDynamicSubject(
+          templateId = "transactionEngineHMRCPSAPRACDEAUTHFailure",
+          fromAddress = transactionEngineAddress,
+          service = ServiceIdentifier.PensionSchemePractitioners,
+          subject = extractSubject(_),
+          plainTemplate = psa.prac.txt.transactionEngineHMRCPSAPRACAUTHFailure.f,
```
Comment (enriquefaci) [inline app/uk/gov/hmrc/hmrcemailrenderer/templates/transactionengine/TransactionEngineTemplates.scala]: Could you check if you are you using the right templates? In these examples the template does not match the html file:

TemplateId is transactionEngineHMRCPSAPRACDEAUTHFailure but you call 
transactionEngineHMRCPSAPRACAUTHFailure.f

TemplateId transactionEngineHMRCPSAPRACDEAUTHSuccess but you call 
transactionEngineHMRCPSAPRACAUTHSuccess.f

File: app/preview/TemplateParams.scala (hunk 1 above)
Comment (yangwong9664) [inline app/preview/TemplateParams.scala]: Hi, they aren't duplicates. We have PRACAUTH and PRACDEAUTH which look the same, but aren't. Also, for each of these two templates we have Failure templates which have two modes, "business" error and a "fatal" error. Therefore, they aren't duplicates they are added to test each scenario. Thanks.

**Comment by yangwong9664** [comment]
Hi @enriquefaci, I've fixed the errors you have found, thanks.

File: app/preview/TemplateParams.scala (hunk 1 above)
Comment (thoeni) [inline app/preview/TemplateParams.scala]: Hi @yangwong9664, I understand that you have different templates, but the `exampleParams` value is a `Map[String, Map[...]]` so if you use the same key twice one will be ignored. It's ok to have it but just use different keys in order to retrieve the desired params from the Map.

Comment (yangwong9664) [inline app/preview/TemplateParams.scala]: Hi @thoeni, the duplicates have been removed, thanks.

**Pull request merged**